src/Aspire.Hosting/EntityFrameworkMigrationExtensions.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// TODO:
+/// </summary>
+public static class EntityFrameworkMigrationExtensions
+{
+    /// <summary>
+    /// TODO
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="builder"></param>
+    /// <param name="projectMetadata"></param>
+    /// <returns></returns>
+    public static IResourceBuilder<T> WithEntityFrameworkMigrations<T>(this IResourceBuilder<T> builder, IProjectMetadata projectMetadata) where T: IResourceSupportsEntityFrameworkMigrations
+    {
+        builder.ApplicationBuilder.AddExecutable($"{builder.Resource.Name}-migrate", "dotnet", ".")
+                                  .WithArgs(context => {
+                                    context.Args.Add("ef");
+                                    context.Args.Add("database");
+                                    context.Args.Add("update");
+                                    context.Args.Add("--project");
+                                    context.Args.Add(projectMetadata.ProjectPath);
+                                    context.Args.Add("--connection");
+                                    context.Args.Add(builder);
+                                  })
+                                  .WaitFor((IResourceBuilder<IResource>)builder);
+        return builder;
```

Comment:
What if your app wants to wait for migrations to be completed?  With thisapproach, your app would need to wait on the migrator resource, but the migrator resource is not returned to the consumer?

An to running `dotnet ef migrate` as a seperate process, you could run the migrations in-process via the `ResourceReadyEvent` - as the event is blocking, consumers could just wait on the database resource directly, although it would require a bit more code to write, wire up logs, handle errors etc.

Reply:
We are considering this option as well.

Reply:
Running it in process mean referencing the DbContext and all of the app code from the apphost, which is a non starter.